test/compounder/AuraBalVault.spec.ts

```diff
@@ -0,0 +1,462 @@
+                const amount = simpleToExactAmount(10);
+                await phase2.cvxCrv.approve(vault.address, amount);
+                await vault.deposit(amount, deployerAddress);
+                await vault.setHarvestPermissions(false);
+
+                // Disable fee token handler
+                await strategy.updateRewardToken(mocks.addresses.feeToken, ZERO_ADDRESS);
+
+                // ----- Send some balance to the strategy to mock the harvest ----- //
+                const { crv } = mocks;
+                const feeToken = MockERC20__factory.connect(mocks.addresses.feeToken, deployer);
+                await crv.transfer(strategy.address, amount);
+                await phase2.cvx.transfer(strategy.address, amount);
+                await feeToken.transfer(strategy.address, amount);
+                // ----- Send some balance to the balancer vault to mock swaps ----- //
+                await phase2.cvxCrv.transfer(mocks.balancerVault.address, amount);
+                await mocks.weth.transfer(mocks.balancerVault.address, amount);
+                await mocks.balancerVault.setTokens(mocks.crvBpt.address, phase2.cvxCrv.address);
+
+                const feeTokenBalance = await feeToken.balanceOf(strategy.address);
+
+                expect(await feeToken.balanceOf(strategy.address), " feeToken balance").to.be.gt(0);
+
+                // Test harvest without fee token handler
+
+                const tx = await vault["harvest(uint256)"](0);
+                await expect(tx).to.emit(vault, "Harvest");
+                // Queue new rewards
+                await expect(tx).to.emit(auraRewards, "RewardAdded");
+                expect(await feeToken.balanceOf(strategy.address), " feeToken balance").to.be.eq(feeTokenBalance);
+            });
+            it("fails if permissioned  and not whitelisted ", async () => {
+                //  Deposit to make sure totalSupply is not ZERO
+                const amount = simpleToExactAmount(10);
+                await phase2.cvxCrv.approve(vault.address, amount);
+                await vault.deposit(amount, deployerAddress);
+                await vault.setHarvestPermissions(true);
+
+                await expect(vault.connect(alice)["harvest()"](), "fails ").to.be.revertedWith("permissioned harvest");
+            });
+        });
+        describe("addExtraReward", async () => {
+            it("fails if caller is not owner", async () => {
+                await expect(
+                    vault.connect(alice).addExtraReward(DEAD_ADDRESS),
+                    "fails due to owner",
+                ).to.be.revertedWith("Ownable: caller is not the owner");
+            });
+            it("fails if wrong address", async () => {
+                await expect(vault.addExtraReward(ZERO_ADDRESS), "fails due to").to.be.revertedWith("Invalid address!");
+            });
+            it("does not add more than 12 rewards", async () => {
```

Comment:
it would be good to check the normal flow with 12 rewards, simply to check it can fit in a block